gradio/blocks.py

```diff
@@ -1099,6 +1119,22 @@ def __init__(
             }
             analytics.initiated_analytics(data)
 
+        node_path = os.environ.get("GRADIO_NODE_PATH", get_node_path())
+        self.node_server_name = None
+        self.node_port = None
+
+        if not Blocks.node_process:
+            (node_server_name, node_process, node_port) = start_node_server(
```

Comment:
Rather than starting node in init, we can do in `startup_events` method of `gr.Blocks`. That way, if you're nesting multiple blocks in a single app, only one node process is started.

And `startup_events` runs even when mounted in a larger FastAPI app.

Reply:
Ok I think we're only going to start one node process anyways with this implementation since we're attaching it to the class but I still think `startup_events` is a bit cleaner.

Reply:
Yeah @abidlabs suggested that but we can't do it there if i remember from my testing because we need to know certain details before mounting the gradio app.

Reply:
Happy to change it in another PR if i'm wrong, i just don't have time to test it atm, the process does only start once tho.

Reply:
i think i might be wrong actually but I'd need to test